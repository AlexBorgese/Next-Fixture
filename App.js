import React from 'react';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';
import { AppLoading, Asset, Font } from 'expo';
import { Ionicons } from '@expo/vector-icons';
import RootNavigation from './src/navigation/RootNavigation';

export default class App extends React.Component {
	state = {
		assetsAreLoaded: false,
	};

	componentWillMount() {
		this._loadAssetsAsync();
	}

	render() {
		if (!this.state.assetsAreLoaded && !this.props.skipLoadingScreen) {
			return <AppLoading />;
		}
			return (
				<View style={styles.container}>
					{Platform.OS === 'ios' && <StatusBar barStyle="default" />}
					{Platform.OS === 'android' &&
						<View style={styles.statusBarUnderlay} />}
					<RootNavigation />
				</View>
			);
		}
	};
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff'
	},
	statusBarUnderlay: {
		height: 24,
		backgroundColor: 'rgba(0,0,0,0.2)'
	}
});
