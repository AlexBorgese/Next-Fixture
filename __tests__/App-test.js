import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';

describe('Given a app component', () => {
	it('should render the loading screen', async () => {
		const component = shallow(<App />);

		expect(component).toMatchSnapshot();
	});

	it('should render the root without loading screen', async () => {
		const component = shallow(<App skipLoadingScreen />);

		expect(component).toMatchSnapshot();
	});
});
