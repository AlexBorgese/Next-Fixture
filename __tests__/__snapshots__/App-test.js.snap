// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a app component should render the loading screen 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <AppLoading />,
  "nodes": Array [
    <AppLoading />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <App />,
    "_instance": App {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": Object {
        "assetsAreLoaded": false,
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <AppLoading />,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;

exports[`Given a app component should render the root without loading screen 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
    }
>
    <StatusBar
        animated={false}
        barStyle="default"
        showHideTransition="fade"
    />
    <RootNavigator />
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                }
      }
>
      <StatusBar
            animated={false}
            barStyle="default"
            showHideTransition="fade"
      />
      <RootNavigator />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <App
      skipLoadingScreen={true}
/>,
    "_instance": App {
      "context": Object {},
      "props": Object {
        "skipLoadingScreen": true,
      },
      "refs": Object {},
      "state": Object {
        "assetsAreLoaded": false,
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <View
      style={
            Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                }
      }
>
      <StatusBar
            animated={false}
            barStyle="default"
            showHideTransition="fade"
      />
      <RootNavigator />
</View>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App
    skipLoadingScreen={true}
/>,
}
`;

exports[`renders the loading screen 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <AppLoading />,
  "nodes": Array [
    <AppLoading />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <App />,
    "_instance": App {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": Object {
        "assetsAreLoaded": false,
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <AppLoading />,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;

exports[`renders the root without loading screen 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
    }
>
    <StatusBar
        animated={false}
        barStyle="default"
        showHideTransition="fade"
    />
    <RootNavigator />
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                }
      }
>
      <StatusBar
            animated={false}
            barStyle="default"
            showHideTransition="fade"
      />
      <RootNavigator />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <App
      skipLoadingScreen={true}
/>,
    "_instance": App {
      "context": Object {},
      "props": Object {
        "skipLoadingScreen": true,
      },
      "refs": Object {},
      "state": Object {
        "assetsAreLoaded": false,
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <View
      style={
            Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                }
      }
>
      <StatusBar
            animated={false}
            barStyle="default"
            showHideTransition="fade"
      />
      <RootNavigator />
</View>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App
    skipLoadingScreen={true}
/>,
}
`;
